mkosi/manifest.py

```diff
@@ -64,6 +64,21 @@ def report(self) -> str:
             t += f"""\nChangelog:\n{self.changelog}\n"""
         return t
 
+def parse_pkg_desc(f: Path) -> List[str]:
+    ret = []
+    with open(f) as desc:
+        lines = [l.strip() for l in desc.readlines()]
+        for n, line in enumerate(lines):
+            if line == "%NAME%":
+                ret.append(lines[n+1])
+            if line == "%VERSION%":
+                ret.append(lines[n+1])
+            if line == "%BASE%":
+                ret.append(lines[n+1])
+            if line == "%ARCH%":
+                ret.append(lines[n+1])
+                break
+    return ret
```

Comment:
```
def parse_pkg_desc(f: Path) -> (str,):
     with f.open() as desc:
        for line in desc:
            match line.strip():
               case '%NAME%':
                  name = next(desc).strip()
               case '%VERSION%':
                  version = next(desc).strip()
               case '%BASE%':
                  base = next(desc).strip()
               case '%ARCH%':
                  arch = next(desc).strip()
                  return name, version, base, arch
```

Untested, but I think that's better.

Reply:
I don't know how reliable those files are. This will "crash" if some field is not found.

Reply:
Using `match` isn't an option until we bump the minimum python version unfortunately

Reply:
I checked if I could use match actually :) but currently `mkosi` is only up to python3.7

Reply:
@keszybz The files should *always* have these fields present, and in *exactly* this order. There is no spec/manpage, and we can make it more flexible. But this is very well handled in `makepkg`

Reply:
We'd need to bump the Python version quite a bit. ;) We're on 3.7+ and `match` has only been available for less than a year with 3.10. I'm looking forward to using it but maybe we should do a more conservative bump before that. 

But I also prefer the pre 3.10 version of the code (among other reasons because the returned type has a more concrete shape).
```python
def parse_pkg_desc(f: Path) -> Tuple[str, str, str, str]:
    with open(f) as desc:
        for line in desc:
            line = line.strip()
            if line == "%NAME%":
                name = next(desc).strip()
            elif line == "%VERSION%":
                version = next(desc).strip()
            elif line == "%BASE%":
                base = next(desc).strip()
            elif line == "%ARCH%":
                arch = next(desc).strip()
                break
    return name, version, base, break
```

Reply:
Ahh, I need to get my Python-foo up to snuff. That is much better.